### PR TITLE
tvutil/ntsc: removed need for empty dummy line

### LINKED ISSUE #58: clerLine() needlessly uses VRAM
Said function utilises half an empty line to efficiently generate empty lines during scrolling. This can be handled differently without much penalty (and getting the VRAM back).

 -  draw a line in background colour at the top of the character line
 -  duplicate this line with h/w blitter (avoid <5 transfer widths, **window 9,0,1,20** looked interesting)
 -  the line drawing will have to be accelerated as it's pixel based which introduces a huge overhead

### PR DESCRIPTION
**Pull request by konimaru**
Fixes #58.

### CODE REVIEW

File: ttbasic/ntsc.cpp [hunk 1]
```diff
@@ -901,16 +904,15 @@ void ICACHE_RAM_ATTR VS23S010::MoveBlock(uint16_t x_src, uint16_t y_src,
 
 	// If the last move was a reverse one, we have to wait until it's
 	// finished before we can set the new addresses.
-	if (last_dir)
-		while (!blockFinished()) {
-		}
+	while (last_dir && !blockFinished()) {}
```
Comment (uli) [inline ttbasic/ntsc.cpp]: This is not good. It makes it unclear what is actually being waited for, and suggests that `last_dir` may be variable within the scope of the `while()` loop.

File: ttbasic/ntsc.cpp [hunk 2]
```diff
@@ -901,16 +904,15 @@ void ICACHE_RAM_ATTR VS23S010::MoveBlock(uint16_t x_src, uint16_t y_src,
...
+
 	SpiRamWriteBMCtrl(BLOCKMVC1, byteaddress2 >> 1, byteaddress1 >> 1,
 			  ((byteaddress1 & 1) << 1) | ((byteaddress2 & 1) << 2)
 			  | dir | lowpass());
-	if (!last_dir)
-		while (!blockFinished()) {
-		}
-	SpiRamWriteBM2Ctrl(m_pitch - width, width, height - 1);
+
+	while (!last_dir && !blockFinished()) {}
```
Comment (uli) [inline ttbasic/ntsc.cpp]: Likewise.

**Comment by konimaru** [comment]
Ok, please hold.

**Comment by konimaru** [comment]
I'm suspending this as it has side-effects in mode 3 and 4 on real h/w. Investigating ...

**Pull request closed without merging**